features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/PrivacyOverrideExtensions.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay
+
+import android.view.View
+
+/**
+ * Allows setting a view to be "hidden" in the hierarchy in Session Replay.
+ * When hidden the view will be replaced with a placeholder in the replay and
+ * no attempt will be made to record it's children.
+ *
+ * @param hide pass `true` to hide the view, or `false` to remove the override
+ */
+fun View.datadogSessionReplayHidden(hide: Boolean) {
+    if (hide) {
```

Comment:
should this be a synchronized block?

Reply:
> [!NOTE]
> Looking at the underlying implementation of the View tags, it's not concurrent safe, especially in the case of setting multiple tags. 
> IMO, I don't feel that it's necessary though, as we're setting a boolean. From the SDK we can only force synchronisation between the write and read, and I feel like the write will happen only once (I can't think of scenarios where we'd have to hide/reveal the same view many times), so it's good enough.

Reply:
Normally, the common rule of working with Views is to call any methods/properties from the main thread (because some methods may throw exception if call wasn't done from the main thread).

However, while customer will probably follow this rule and call `setTag`, `getTag` on our side may happen on the worker thread I suppose?

Reply:
We are calling get only on the main thread, but customer could still theoretically call set from other threads. IAC it seems synchronization will not be necessary here